src/Uno.UI/UI/Xaml/Internal/RootVisual.cs

```diff
@@ -98,5 +102,42 @@ protected override Size ArrangeOverride(Size finalSize)
 
 			return finalSize;
 		}
+
+
+#if HAS_UNO
+		// Uno specific: To ensure focus is properly lost when clicking "outside" app's content,
+		// we set focus here. In case UWP, focus is set to the root ScrollViewer instead,
+		// but Uno does not have it on all targets yet.
+		private bool _isLeftButtonPressed = false;
+
+		private void RootVisual_PointerPressed(object sender, Windows.UI.Xaml.Input.PointerRoutedEventArgs e)
+		{
+			if (e.Handled)
+			{
+				return;
+			}
+
+			var point = e.GetCurrentPoint(this);
+			_isLeftButtonPressed = point.Properties.IsLeftButtonPressed;
+		}
+
+		private void RootVisual_PointerReleased(object sender, Windows.UI.Xaml.Input.PointerRoutedEventArgs e)
+		{
+			if (e.Handled)
+			{
+				return;
+			}
+			if (_isLeftButtonPressed)
+			{
+				_isLeftButtonPressed = false;
```

Comment:
Should be handle `PointerCanceled` as well ? /cc @dr1rrb

Reply:
Good thinking, will adjust this

Reply:
+1 (presumably you want to reset the field without actually unfocusing)

Reply:
Done